src/ansys/additive/core/logger.py

```diff
@@ -0,0 +1,286 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Provides a general framework for logging in PyAdditive.
+
+Logger usage
+------------
+
+Global logger
+~~~~~~~~~~~~~
+There is a global logger named ``PyAdditive_global`` that is created when
+``ansys.additive.core.__init__`` is called. If you want to use this global
+logger, you must call it at the top of your module:
+
+.. code:: python
+
+   from ansys.additive.core import LOG
+
+You can rename this logger to avoid conflicts with other loggers (if any):
+
+.. code:: python
+
+   from ansys.additive.core import LOG as logger
+
+The default logging level of ``LOG`` is ``ERROR``.
+You can change this level and output lower-level messages with
+this code:
+
+.. code:: python
+
+   LOG.logger.setLevel("DEBUG")
+   LOG.file_handler.setLevel("DEBUG")  # If present.
+   LOG.stdout_handler.setLevel("DEBUG")  # If present.
+
+Alternatively, you can ensure that all the handlers are set to the input log
+level with this code:
+
+.. code:: python
+
+   LOG.setLevel("DEBUG")
+
+This logger does not log to a file by default. If you want, you can
+add a file handler with this code:
+
+.. code:: python
+
+   import os
+
+   file_path = os.path.join(os.getcwd(), "pyadditive.log")
+   LOG.log_to_file(file_path)
+"""
```

Comment:
Add @julieatansys to the reviewers. She may want to change some of the text above.